base/test.jl

```diff
@@ -908,4 +908,11 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+immutable GenericString <: AbstractString
+    string::AbstractString
```

Comment:
I also plan on making this type "lumpy" – i.e. each character will take a random number of bytes. That way tests that use this type will really put code through its paces in terms of assumptions about encodings and such.